### PR TITLE
ready for submission

### PR DESCRIPTION
**Pull request by eshaan-mehta**

### CODE REVIEW

File: modules/bootcamp/detect_landing_pad.py [hunk 1]
```diff
@@ -84,33 +85,43 @@ def run(self, image: np.ndarray) -> "tuple[list[bounding_box.BoundingBox], np.nd
...
+        
 
         # Get the Result object
-        prediction = ...
+        prediction = predictions[0]
 
         # Plot the annotated image from the Result object
         # Include the confidence value
-        image_annotated = ...
+        image_annotated = prediction.plot(conf=self.__CONFIDENCE_THRESHOLD)
```
Comment (Xierumeng) [inline modules/bootcamp/detect_landing_pad.py:99]: Is this how `plot()` is supposed to be used?

File: modules/bootcamp/detect_landing_pad.py [hunk 2]
```diff
@@ -84,33 +85,43 @@ def run(self, image: np.ndarray) -> "tuple[list[bounding_box.BoundingBox], np.nd
...
 
         # Get the xyxy boxes list from the Boxes object in the Result object
-        boxes_xyxy = ...
+        boxes_xyxy = prediction.boxes.xyxy
 
         # Detach the xyxy boxes to make a copy,
         # move the copy into CPU space,
         # and convert to a numpy array
-        boxes_cpu = ...
+        boxes_cpu = boxes_xyxy.cpu().numpy()
```
Comment (Xierumeng) [inline modules/bootcamp/detect_landing_pad.py:107]: Should detach before copying to the CPU.

File: modules/bootcamp/detect_landing_pad.py [hunk 3]
```diff
@@ -84,33 +85,43 @@ def run(self, image: np.ndarray) -> "tuple[list[bounding_box.BoundingBox], np.nd
...
+                bounding_boxes.append(box)
+
         # Hint: .shape gets the dimensions of the numpy array
         # for i in range(0, ...):
             # Create BoundingBox object and append to list
             # result, box = ...
 
         # Remove this when done
-        raise NotImplementedError
+        #raise NotImplementedError
```
Comment (Xierumeng) [inline modules/bootcamp/detect_landing_pad.py:122]: Remove this line.

File: modules/bootcamp/detect_landing_pad.py [hunk 4]
```diff
@@ -84,33 +85,43 @@ def run(self, image: np.ndarray) -> "tuple[list[bounding_box.BoundingBox], np.nd
...
         # and convert to a numpy array
-        boxes_cpu = ...
+        boxes_cpu = boxes_xyxy.cpu().numpy() 
 
         # Loop over the boxes list and create a list of bounding boxes
         bounding_boxes = []
+        for i in range(0, len(boxes_cpu)):
+            result, box = bounding_box.BoundingBox.create(boxes_cpu[i])
+            if result:
+                bounding_boxes.append(box)
```
Comment (Xierumeng) [inline modules/bootcamp/detect_landing_pad.py:114]: In statistics, an invalid portion of the data point involves removing the entirety of the data point (where the data point here is an image, as each image is independent). (You do not need to change anything here, just something to think about).

File: modules/bootcamp/decision_simple_waypoint.py [hunk 5]
```diff
@@ -11,7 +11,7 @@
 from .. import drone_report
 from .. import drone_status
 from .. import location
-from ..private.decision import base_decision
+from .. private.decision import base_decision
```
Comment (Xierumeng) [inline modules/bootcamp/decision_simple_waypoint.py:14]: ???

File: modules/bootcamp/decision_simple_waypoint.py [hunk 6]
```diff
@@ -36,13 +36,14 @@ def __init__(self, waypoint: location.Location, acceptance_radius: float):
         # ============
         # ↓ BOOTCAMPERS MODIFY BELOW THIS COMMENT ↓
         # ============
-
-        # Add your own
+
```
Comment (Xierumeng) [inline modules/bootcamp/decision_simple_waypoint.py:39]: Avoid leaving unnecessary whitespace.

File: modules/bootcamp/decision_simple_waypoint.py [hunk 7]
```diff
@@ -68,13 +69,29 @@ def run(self,
         # ↓ BOOTCAMPERS MODIFY BELOW THIS COMMENT ↓
         # ============
 
-        # Do something based on the report and the state of this class...
+        # Do something based on the report and the state of this class...\
+       
+        # actions for once drone has reached destination
+        if report.position.__eq__(report.destination) and self.has_taken_off:
```
Comment (Xierumeng) [inline modules/bootcamp/decision_simple_waypoint.py:75]: Do not use private methods outside the scope they are declared in! Private methods are denoted with a double underscore prefix.

Comment (Xierumeng) [inline modules/bootcamp/decision_simple_waypoint.py:75]: Avoid equality comparison with floating point values, since there might be some tiny difference (e.g. 0.0000001) which will cause it to fail. Better to use a difference and see whether it is within a tolerance (acceptance radius perhaps?).

File: modules/bootcamp/decision_simple_waypoint.py [hunk 8]
```diff
@@ -68,13 +69,29 @@ def run(self,
...
+        # set destination to waypoint before drone takes off, and set take off status
+        # to allow setting destination to waypoint before checking 
+        if not self.has_taken_off:
+                command = commands.Command.create_set_relative_destination_command(self.waypoint.location_x - report.position.location_x, self.waypoint.location_y  - report.position.location_y)
+                self.has_taken_off = True
+        
 
         # Remove this when done
-        raise NotImplementedError
+        #raise NotImplementedError
```
Comment (Xierumeng) [inline modules/bootcamp/decision_simple_waypoint.py:90]: Remove.

File: modules/bootcamp/decision_simple_waypoint.py [hunk 9]
```diff
@@ -68,13 +69,29 @@ def run(self,
...
 
         # ============
         # ↑ BOOTCAMPERS MODIFY ABOVE THIS COMMENT ↑
         # ============
 
         return command
+
```
Comment (Xierumeng) [inline modules/bootcamp/decision_simple_waypoint.py:97]: ???

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 10]
```diff
@@ -38,10 +38,35 @@ def __init__(self, waypoint: location.Location, acceptance_radius: float):
...
         # ============
         # ↑ BOOTCAMPERS MODIFY ABOVE THIS COMMENT ↑
         # ============
+    def find_nearest_landing_pad(self, waypoint: location.Location, landing_pad_locations: "list[location.Location]") -> int:
+        """
+        To determine closest landing pad in frame to waypoint and return it's index.
+        Index is returned instead of object to improve speed
+        """
+        
+        # setting intial shortest values, to be changed
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:55]: To be changed?

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 11]
```diff
@@ -38,10 +38,35 @@ def __init__(self, waypoint: location.Location, acceptance_radius: float):
...
+        shortest_distance = 10000000
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:56]: Would this work if distances were measured in micrometres? Is there a value that is guaranteed to be larger than every other value?

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 12]
```diff
@@ -38,10 +38,35 @@ def __init__(self, waypoint: location.Location, acceptance_radius: float):
...
+        shortest_index = None
+
+        for i, pad in enumerate(landing_pad_locations):
+            # use distance formula to find distance between pad and waypoint
+            distance = ((waypoint.location_x - pad.location_x)**2 + (waypoint.location_y - pad.location_y)**2)**0.5
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:61]: Square roots are computationally expensive. Is there a way to avoid using it? (You do not need to change anything here, just something to think about).

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 13]
```diff
@@ -38,10 +38,35 @@ def __init__(self, waypoint: location.Location, acceptance_radius: float):
...
         # Add your own
+        self.destination = waypoint #preset destination to waypoint so drone has intial direction
+        self.has_taken_off = False
+        self.change_destination = False
+        self.prev_num_landing_pads = 1
 
         # ============
         # ↑ BOOTCAMPERS MODIFY ABOVE THIS COMMENT ↑
         # ============
+    def find_nearest_landing_pad(self, waypoint: location.Location, landing_pad_locations: "list[location.Location]") -> int:
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:49]: Good use of type hints!

Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:49]: This could be a static method.

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 14]
```diff
@@ -67,11 +92,47 @@ def run(self,
         # ============
         # ↓ BOOTCAMPERS MODIFY BELOW THIS COMMENT ↓
         # ============
-
+        
         # Do something based on the report and the state of this class...
+        
+        
+        #print(self.change_destination, self.has_taken_off, report.status == drone_status.DroneStatus.MOVING)
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:99]: Remove.

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 15]
```diff
@@ -67,11 +92,47 @@ def run(self,
...
+        
+        # actions for once drone has taken off and is flying
+        
+        if self.has_taken_off:
+            
+            # update destination to closest landing pad to waypoint
+            # only updates closest landing pad once there is a change in the number of landing pads available
+            # prevents drone from going back to origin, looking for landing pads when none are visible, and repeatedly searching for new closest when same pads are on screen
+            num_landing_pads = len(landing_pad_locations)
+            if num_landing_pads > 0 and not self.prev_num_landing_pads == num_landing_pads:
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:109]: What if there is a landing pad between the initial position and the waypoint that is far from the waypint, and another landing pad near the waypoint? The camera would see the far landing pad and set it as the destination. The camera may not see the near waypoint (if it's on the other side of the waypoint).

Additionally, the task requires the drone to reach the waypoint before going to the nearest landing pad. Does the drone actually reach the waypoint every time?

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 16]
```diff
@@ -67,11 +92,47 @@ def run(self,
...
+                if report.status == drone_status.DroneStatus.MOVING:  # halt if previously moving
+                    command = commands.Command.create_halt_command()
+                if report.status == drone_status.DroneStatus.HALTED:  # land if previously halted, and prepare status for next takeoff
+                    command = commands.Command.create_land_command()
+                    self.has_taken_off = False
+            
+                
+        # sets drones initial destination to waypoint and destinations to landing pads as they become visible
+        # updates drone flight and direction changing status
+        if report.status == drone_status.DroneStatus.HALTED and (not self.has_taken_off or self.change_destination) and not report.position.__eq__(self.destination):
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:126]: Sometimes it is okay to write duplicate code, to avoid hard to read conditions like this.

File: modules/bootcamp/decision_simple_waypoint.py (hunk 5 above)
Comment (eshaan-mehta) [reply modules/bootcamp/decision_simple_waypoint.py:14]: I just added a space there. my ocd was kicking it lol

File: modules/bootcamp/decision_waypoint_landing_pads.py (hunk 13 above)
Comment (eshaan-mehta) [reply modules/bootcamp/decision_waypoint_landing_pads.py:49]: could you explain what that means. I am unfamiliar with static methods. Thanks!

File: modules/bootcamp/detect_landing_pad.py (hunk 2 above)
Comment (eshaan-mehta) [reply modules/bootcamp/detect_landing_pad.py:107]: could you explain fully what it means to detach?

File: modules/bootcamp/detect_landing_pad.py (hunk 1 above)
Comment (eshaan-mehta) [reply modules/bootcamp/detect_landing_pad.py:99]: This is how it was used in the ultralytics documentation. Could you explain further?

File: modules/bootcamp/decision_waypoint_landing_pads.py (hunk 15 above)
Comment (eshaan-mehta) [reply modules/bootcamp/decision_waypoint_landing_pads.py:109]: oh my bad. I thought it was supposed to go directly to the nearest landing pad. So when it detects the closest one, it changes course and head straight there. Thank you for the clarification

File: modules/bootcamp/detect_landing_pad.py [hunk 17]
```diff
@@ -84,33 +85,41 @@ def run(self, image: np.ndarray) -> "tuple[list[bounding_box.BoundingBox], np.nd
...
+        
 
         # Get the Result object
-        prediction = ...
+        prediction = predictions[0]
 
         # Plot the annotated image from the Result object
         # Include the confidence value
-        image_annotated = ...
+        image_annotated = prediction.plot()
```
Comment (Xierumeng) [inline modules/bootcamp/detect_landing_pad.py:99]: Do the output images have the confidence labels?

File: modules/bootcamp/detect_landing_pad.py (hunk 2 above)
Comment (Xierumeng) [reply modules/bootcamp/detect_landing_pad.py]: There is a method that detaches the Pytorch tensor.

File: modules/bootcamp/detect_landing_pad.py [hunk 18]
```diff
@@ -84,33 +85,41 @@ def run(self, image: np.ndarray) -> "tuple[list[bounding_box.BoundingBox], np.nd
...
 
         # Get the xyxy boxes list from the Boxes object in the Result object
-        boxes_xyxy = ...
+        boxes_xyxy = prediction.boxes.xyxy
 
         # Detach the xyxy boxes to make a copy,
         # move the copy into CPU space,
         # and convert to a numpy array
-        boxes_cpu = ...
+        boxes_cpu = boxes_xyxy.cpu()
```
Comment (Xierumeng) [inline modules/bootcamp/detect_landing_pad.py:107]: https://github.com/UWARG/autonomy-bootcamp-2023/pull/10#discussion_r1327901224

File: modules/bootcamp/decision_simple_waypoint.py (hunk 5 above)
Comment (Xierumeng) [reply modules/bootcamp/decision_simple_waypoint.py:14]: Do not add a space. The import path is specifically `..private.decision` .

File: modules/bootcamp/decision_simple_waypoint.py (hunk 6 above)
Comment (Xierumeng) [reply modules/bootcamp/decision_simple_waypoint.py:39]: Please address.

File: modules/bootcamp/decision_simple_waypoint.py [hunk 19]
```diff
@@ -36,13 +36,27 @@ def __init__(self, waypoint: location.Location, acceptance_radius: float):
...
-
-        # Add your own
+        
+        self.has_taken_off = False
 
         # ============
         # ↑ BOOTCAMPERS MODIFY ABOVE THIS COMMENT ↑
         # ============
 
+    def has_arrived(self, postition: location.Location, destination: location.Location) -> bool:
```
Comment (Xierumeng) [inline modules/bootcamp/decision_simple_waypoint.py:46]: Good use of type hints!

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 20]
```diff
@@ -38,11 +38,54 @@ def __init__(self, waypoint: location.Location, acceptance_radius: float):
         # ============
 
         # Add your own
+        self.destination = self.waypoint #preset destination to waypoint so drone has intial direction
+        self.has_taken_off = False
+        #self.change_destination = False
+        #self.prev_num_landing_pads = 1
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:44]: Remove these if unused. Don't leave commented code lying around.

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 21]
```diff
@@ -38,11 +38,54 @@ def __init__(self, waypoint: location.Location, acceptance_radius: float):
...
+
+
+    def find_nearest_landing_pad(self, waypoint: location.Location, landing_pad_locations: "list[location.Location]") -> int:
+        """
+        To determine closest landing pad in frame to waypoint and return it's index.
+        Index is returned instead of object to improve speed
+        """
+        
+        # setting intial shortest values
+        shortest_distance = float('inf')
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:72]: Use double quotation marks for string literals.

File: modules/bootcamp/decision_waypoint_landing_pads.py (hunk 13 above)
Comment (Xierumeng) [reply modules/bootcamp/decision_waypoint_landing_pads.py:65]: https://pythonbasics.org/static-method/

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 22]
```diff
@@ -67,11 +110,60 @@ def run(self,
...
         # Do something based on the report and the state of this class...
-
-        # Remove this when done
-        raise NotImplementedError
+        
+        # actions for once drone has reached destination
+        if self.has_arrived(report.position, report.destination) and self.has_taken_off:
+            if report.status == drone_status.DroneStatus.MOVING:  # halt if previously moving
+                command = commands.Command.create_halt_command()
+            if report.status == drone_status.DroneStatus.HALTED:  # actions once halted
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:120]: What if the drone could stop for some reason in the middle of a movement? (You do not need to change anything here, just something to think about).

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 23]
```diff
@@ -67,11 +110,60 @@ def run(self,
...
         # ============
-
+        
         # Do something based on the report and the state of this class...
-
-        # Remove this when done
-        raise NotImplementedError
+        
+        # actions for once drone has reached destination
+        if self.has_arrived(report.position, report.destination) and self.has_taken_off:
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:117]: What if the drone could halt for some reason in the middle of a movement? (You do not need to change anything here, just something to think about).

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 24]
```diff
@@ -67,11 +110,60 @@ def run(self,
...
+            if report.status == drone_status.DroneStatus.MOVING:  # halt if previously moving
+                command = commands.Command.create_halt_command()
+            if report.status == drone_status.DroneStatus.HALTED:  # actions once halted
+                # if at waypoint, find nearest landing pad and set new destination
+                # otherwise land, since drone will be at landing pad
+                if self.destination == self.waypoint:
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:123]: Better to avoid the equality operator for floating point values (what if it was off by 0.00001?). You could take the difference and make sure it's under some tolerance (acceptance radius perhaps?).

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 25]
```diff
@@ -67,11 +110,60 @@ def run(self,
...
+            
+                
+        # # sets drones initial destination to waypoint and destinations to landing pads as they become visible
+        # # updates drone flight and direction changing status
+        # if report.status == drone_status.DroneStatus.HALTED and (not self.has_taken_off or self.change_destination) and not self.has_arrived(report.position, self.destination):
+        #     command = commands.Command.create_set_relative_destination_command(self.destination.location_x - report.position.location_x, self.destination.location_y - report.position.location_y)
+        #     self.has_taken_off = True
+        #     self.change_destination = False
+
+        # self.prev_num_landing_pads = len(landing_pad_locations)  # update number of landing pads from the previous frame
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:165]: Remove unused code.

File: modules/bootcamp/decision_waypoint_landing_pads.py (hunk 21 above)
Comment (eshaan-mehta) [reply modules/bootcamp/decision_waypoint_landing_pads.py:72]: they are double quotations

File: modules/bootcamp/decision_simple_waypoint.py (hunk 5 above)
Comment (eshaan-mehta) [reply modules/bootcamp/decision_simple_waypoint.py:14]: sorry about that

Comment (Xierumeng) [inline modules/bootcamp/decision_simple_waypoint.py:14]: Please remove the space.

File: modules/bootcamp/decision_waypoint_landing_pads.py (hunk 21 above)
Comment (Xierumeng) [reply modules/bootcamp/decision_waypoint_landing_pads.py:56]: Double quotation mark: `"`

Single quotation mark: `'`

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 26]
```diff
@@ -38,10 +38,49 @@ def __init__(self, waypoint: location.Location, acceptance_radius: float):
...
+        Index is returned instead of object to improve speed
+        """
+        
+        # setting intial shortest values
+        shortest_distance = float('inf')
+        shortest_index = None 
+
+        for i, pad in enumerate(landing_pad_locations):
+            # use distance formula to find distance between pad and waypoint
+            distance = float(((waypoint.location_x - pad.location_x)**2 + (waypoint.location_y - pad.location_y)**2)**0.5)
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:61]: Cast to float is unnecessary.

File: modules/bootcamp/decision_waypoint_landing_pads.py (hunk 21 above)
Comment (eshaan-mehta) [reply modules/bootcamp/decision_waypoint_landing_pads.py:56]: oh whoops, for some reason, i though you were talking about the function description. My bad. Fixed

File: modules/bootcamp/decision_simple_waypoint.py (hunk 5 above)
Comment (eshaan-mehta) [reply modules/bootcamp/decision_simple_waypoint.py]: fixed

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 27]
```diff
@@ -67,11 +106,27 @@ def run(self,
...
+        # actions for once drone has reached destination
+        if self.is_same(report.position, report.destination) and self.has_taken_off:
+            if report.status == drone_status.DroneStatus.MOVING:  # halt if previously moving
+                command = commands.Command.create_halt_command()
+            if report.status == drone_status.DroneStatus.HALTED:  # actions once halted
+                # if at waypoint, find nearest landing pad and set new destination
+                # otherwise land, since drone will be at landing pad
+                if self.is_same(self.destination, self.waypoint):
+                    index = self.find_nearest_landing_pad(self.waypoint, landing_pad_locations)
+                    self.destination = landing_pad_locations[index]
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:121]: What if the closest landing pad is on top of the waypoint?

File: modules/bootcamp/decision_waypoint_landing_pads.py (hunk 26 above)
Comment (eshaan-mehta) [reply modules/bootcamp/decision_waypoint_landing_pads.py]: right. Fixed

File: modules/bootcamp/decision_waypoint_landing_pads.py (hunk 27 above)
Comment (eshaan-mehta) [reply modules/bootcamp/decision_waypoint_landing_pads.py:121]: I changed it from self.destination to report.destination and I believe that fixes the issue as report.destination doesn't get update until the next frame

File: modules/bootcamp/tests/1694876877_results.txt
Comment (Xierumeng) [inline modules/bootcamp/tests/1694876877_results.txt:1]: How did this get here?

File: modules/bootcamp/test.py
Comment (Xierumeng) [inline modules/bootcamp/test.py:1]: Remove.

File: modules/bootcamp/decision_simple_waypoint.py [hunk 28]
```diff
@@ -68,10 +82,21 @@ def run(self,
...
         # ============
 
-        # Do something based on the report and the state of this class...
-
-        # Remove this when done
-        raise NotImplementedError
+        # Do something based on the report and the state of this class...\
+       
+        # actions for once drone has reached destination
+        if self.is_same(report.position, report.destination) and self.has_taken_off:
```
Comment (Xierumeng) [inline modules/bootcamp/decision_simple_waypoint.py:88]: What if it was possible that the drone halts in the middle of a movement? (You do not need to change anything here, just something to think about).

File: modules/bootcamp/decision_waypoint_landing_pads.py (hunk 27 above)
Comment (Xierumeng) [reply modules/bootcamp/decision_waypoint_landing_pads.py:121]: Are you sure? Try temporarily changing line 126 of `run_decision_waypoint_landing_pads.py` to `landing_pad_locations = [waypoint]` and running the simulation. Does it work?

Comment (eshaan-mehta) [reply modules/bootcamp/decision_waypoint_landing_pads.py:122]: I see what you mean. I made a quick addition and I believe it should be good now

File: modules/bootcamp/detect_landing_pad.py [hunk 29]
```diff
@@ -84,33 +85,40 @@ def run(self, image: np.ndarray) -> "tuple[list[bounding_box.BoundingBox], np.nd
...
+
         # Hint: .shape gets the dimensions of the numpy array
         # for i in range(0, ...):
             # Create BoundingBox object and append to list
             # result, box = ...
 
-        # Remove this when done
-        raise NotImplementedError
-
+        return (bounding_boxes, image_annotated)
```
Comment (Xierumeng) [inline modules/bootcamp/detect_landing_pad.py:121]: Remove unnecessary parantheses.

**Comment by Xierumeng** [comment]
Please remove extra whitespace (e.g. spaces on empty lines).

Please also remove extra files `1694876877_results.txt` and `test.py` .

**Comment by eshaan-mehta** [comment]
> Please remove extra whitespace (e.g. spaces on empty lines).
> 
> Please also remove extra files `1694876877_results.txt` and `test.py` .

File: modules/bootcamp/decision_simple_waypoint.py [hunk 30]
```diff
@@ -36,13 +36,27 @@ def __init__(self, waypoint: location.Location, acceptance_radius: float):
         # ============
         # ↓ BOOTCAMPERS MODIFY BELOW THIS COMMENT ↓
         # ============
-
-        # Add your own
+
```
Comment (Xierumeng) [inline modules/bootcamp/decision_simple_waypoint.py:39]: Remove whitespace.

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 31]
```diff
@@ -67,11 +107,28 @@ def run(self,
         # ============
         # ↓ BOOTCAMPERS MODIFY BELOW THIS COMMENT ↓
         # ============
-
+
```
Comment (Xierumeng) [inline modules/bootcamp/decision_waypoint_landing_pads.py:110]: Remove whitespace.

Comment (eshaan-mehta) [reply modules/bootcamp/decision_waypoint_landing_pads.py:110]: I keep trying to remove it but when I try to commit my changes, git doesn't consider it a change so it doesn't update anything

Comment (eshaan-mehta) [reply modules/bootcamp/decision_waypoint_landing_pads.py]: I tried it again. Can you see the changes?

Comment (Xierumeng) [reply modules/bootcamp/decision_waypoint_landing_pads.py]: Yes.

File: modules/bootcamp/detect_landing_pad.py (hunk 29 above)
Comment (Xierumeng) [reply modules/bootcamp/detect_landing_pad.py:121]: Reminder.

Comment (eshaan-mehta) [reply modules/bootcamp/detect_landing_pad.py]: whoops. I made the change on my end just forgot to dave the file before commiting

**Review by Xierumeng — COMMENTED** [review]
LGTM.
